lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -170,8 +170,11 @@ def add_generated_directed_dependency(parent, child, label=nil)
     }
 
     if not edge_exists
+      parameters = target.original_parameters
+      resource = ::Puppet::Resource.new(target.class, target.title, :parameters => parameters)
```

Comment:
Style (I think) nit, but is there any reason not to forgo the local and do something like:

```
      resource = ::Puppet::Resource.new(target.class, target.title, 
                                        :parameters => target.original_parameters)
```